Source/1.3/AI/AIFacilityManager.cs

```diff
@@ -50,111 +43,102 @@ public HashSet<FacilityDef> FacilityDefsInstalled
         {
             get
             {
-                if(cachedFacilitiesDef.EnumerableNullOrEmpty() || updateDefCache)
+                if (cachedFacilityDefs.EnumerableNullOrEmpty() || updateDefCache)
                 {
-                    updateDefCache= false;
-                    foreach(FacilityManager manager in Facilities)
+                    updateDefCache = false;
+                    foreach (FacilityManager manager in Facilities)
                     {
-                        cachedFacilitiesDef.AddRange(manager.FacilityDefsInsalled);
+                        cachedFacilityDefs.AddRange(manager.FacilityDefsInstalled);
                     }
                 }
-                return cachedFacilitiesDef;
+
+                return cachedFacilityDefs;
             }
         }
 
         /// <summary>
-        /// Find a low resource and attempt to build a facility for it.
+        ///     Find a low resource and attempt to build a facility for it.
         /// </summary>
-        /// <returns></returns>
+        /// <returns>Whether a <see cref="Facility" /> was successfully built</returns>
         public bool BuildResourceFacility()
         {
             List<ResourceDef> resourceDefs = player.ResourceManager.FindLowResources();
             ResourceDef def = resourceDefs.RandomElement();
 
-            if (FacilityDefsInstalled.Any(x=>x.ProducedResources.Contains(def)))
-            {
-                FacilityDef facilityDef = FacilityDefsInstalled.Where(x=>x.ProducedResources.Contains(def)).RandomElement();
-                return BuildNewFacility(facilityDef);
-            }
-            return false;
+            FacilityDef facilityDef = FacilityDefsInstalled.Where(facility => facility.ProducedResources.Contains(def)).RandomElementWithFallback();
+
+            return facilityDef != null && BuildNewFacility(facilityDef);
         }
 
 
         /// <summary>
-        /// Attempt to build a facility
+        ///     Attempt to build a facility
         /// </summary>
-        /// <param name="facility"></param>
+        /// <param name="facilityDef"></param>
         /// <returns></returns>
         public bool BuildNewFacility(FacilityDef facilityDef)
         {
-            bool hasRemovedAll = true;
+            var hasRemovedAll = true;
```

Comment:
saves a single character => is that really necessary?

Reply:
It just doesn't feel very useful in this case. that's what I mean

Reply:
neither option is more useful than the other here, really, but it's consistent this way